### PR TITLE
Add release notes for v2.3.4

### PR DESCRIPTION
**Pull request by chvck**

### CODE REVIEW

File: modules/project-docs/pages/sdk-release-notes.adoc [hunk 1]
```diff
@@ -28,6 +28,31 @@ Older versions may work, but are not supported.
 
 The most current and up to date API Documentation is always available through the https://pkg.go.dev/github.com/couchbase/gocb/v2[godoc website].
 
+== Version 2.3.4 (16 November 2021)
+
+Version 2.3.4 is a maintenance release for the Go SDK 2.3.0.
+
+https://pkg.go.dev/github.com/couchbase/gocb/v2@v2.3.4?tab=doc[API Documentation]
```
Comment (osfameron) [inline modules/project-docs/pages/sdk-release-notes.adoc:35]: Currently 404

Comment (chvck) [reply modules/project-docs/pages/sdk-release-notes.adoc:35]: Try again, think the pkg.go.dev index just needed poking.

**Comment by osfameron** [comment]
OK, looks good! Can you glance over my gardening in case any issues?

File: modules/project-docs/pages/sdk-release-notes.adoc [hunk 2]
```diff
@@ -28,6 +28,39 @@ Older versions may work, but are not supported.
...
+
+* https://issues.couchbase.com/browse/GOCBC-1186[GOCBC-1186]:
+Fixed issue where logging meter could cause a deadlock on closing the cluster object.
+
+* https://issues.couchbase.com/browse/GOCBC-1187[GOCBC-1187]:
+Fixed issue where logging meter could log a service/operation pair which has no operations.
+
+* https://issues.couchbase.com/browse/GOCBC-1194[GOCBC-1194]:
+Changed ordering of route config bootstrapping, to check all seed nodes for the default network type first.
+This fixed an issue with stuck deployments after upgrade to server 7.0.2.
```
Comment (chvck) [inline modules/project-docs/pages/sdk-release-notes.adoc]: Unsure we should include this line?

**Comment by chvck** [comment]
> OK, looks good! Can you glance over my gardening in case any issues?

Looks good to me other than the one comment. Much improved upon my notes :D

File: modules/project-docs/pages/sdk-release-notes.adoc (hunk 2 above)
Comment (osfameron) [reply modules/project-docs/pages/sdk-release-notes.adoc]: modified after consultation with @RichardSmedley

**Review by osfameron — APPROVED** [review]
👍

**Pull request merged**